sdk/resourcemanager/Azure.ResourceManager.Core/src/ContainerBase.cs

```diff
@@ -48,6 +48,14 @@ protected ContainerBase(ResourceOperationsBase parent)
         /// </summary>
         protected ResourceOperationsBase Parent { get; }
 
+        /// <summary>
+        /// The typed resource identifier for the underlying resource container.
+        /// </summary>
+        public virtual new TIdentifier Id
+        {
+            get { return base.Id as TIdentifier; }
```

Comment:
`base.Id` is the Id of the **parent** resource, while `TIdentifier` is the Id of **this** resource, so this is not a good idea.